### ISSUE TITLE
fireball.cpp fails to build

### ISSUE THREAD

**Issue by Quix0r** [dxx-rebirth/dxx-rebirth#66]
The mentioned file fails to build on Debian:

````
In file included from common/main/object.h:41:0,
                 from similar/main/fireball.cpp:38:
In static member function ‘static T valbaseptridxutil_t<P, I>::check_null_pointer(const A&, T) [with A = object_array_t; T = const object*; P = const object; I = short int]’,
    inlined from ‘static T valbaseptridxutil_t<P, I>::check_null_pointer(T) [with T = const object*; P = const object; I = short int]’ at common/include/valptridx.h:115:65,
    inlined from ‘valptr_t<P, I>::operator valptr_t<P, I>::cpointer_type() const [with P = const object; I = short int; valptr_t<P, I>::cpointer_type = const object*]’ at common/include/valptridx.h:255:121,
    inlined from ‘vvalptr_t<P, I>::vvalptr_t(const valptr_t<const P, I>&) [with P = const object; I = short int]’ at common/include/valptridx.h:518:55,
    inlined from ‘vcobjptr_t::vcobjptr_t(Args&& ...) [with Args = {const cobjptridx_t&}]’ at common/main/object.h:426:1,
    inlined from ‘objptridx_t _ZL27object_create_explosion_sub11objptridx_t12vsegptridx_tRK10vms_vectoriiiii12cobjptridx_t.constprop.70(objptridx_t, vsegptridx_t, const vms_vector&, fix, cobjptridx_t)’ at similar/main/fireball.cpp:218:59,
    inlined from ‘objptridx_t _Z30object_create_badass_explosion11objptridx_t12vsegptridx_tRK10vms_vectoriiiii12cobjptridx_t.constprop.69(objptridx_t, vsegptridx_t, const vms_vector&, fix, cobjptridx_t)’ at similar/main/fireball.cpp:290:138,
    inlined from ‘void do_exploding_wall_frame()’ at similar/main/fireball.cpp:1352:7:
common/include/valptridx.h:120:4: error: call to ‘dxx_trap_null_pointer_exception’ declared with attribute error: NULL pointer used
distcc[26773] ERROR: compile similar/main/fireball.cpp on localhost failed
scons: *** [similar/main/.d2x-rebirth.fireball.o] Error 1
scons: building terminated because of errors.
````

gcc version 4.9.1 (Debian 4.9.1-15)

**Comment by vLKp** [comment]
This works for me using `gcc version 4.9.2` with commit 38e8967fbaac780cfe6c6aa755b00e0922d46427 both with and without `-DEDITOR`.  Based on the callstack, this is the implicit construction of `vcobjptr_t who` from `parent`, which would be `object_none` for the shown callstack.

**Comment by vLKp** [comment]
This should be fixed by bfeddc35014ecde01d2e3ee3e3589669e704b6f0.

**Comment by Quix0r** [comment]
Yes, it is compiling now. Thank you. :-)
